### ISSUE TITLE
Possible mistake in Context Processing Algorithm

### ISSUE THREAD

**Issue by timothee-haudebourg** [w3c/json-ld-api#518]
Hi,

Step 5.5.1 of the Context Processing Algorithm says:
> If `override protected` is false and `active context` contains any protected term definitions, an `invalid context nullification` has been detected and processing is aborted.

The verification is done on `active context` (the input context of the algorithm) and not `result` (the context being built). as far as I understand, this means that the following context, given in [compact#tpr01](https://w3c.github.io/json-ld-api/tests/compact-manifest.html#tpr01) should be accepted:
```json
{
  "@context": [
    {
      "@vocab": "http://example.com/",
      "@version": 1.1,
      "protected": {"@protected": true}
    },
    null
  ]
}
```
...but it is not. An `invalid context nullification` is expected to be returned. So my conclusion is that in Step 5.5.1 of the processing algorithm, `result` must be used to check the presence of protected term definitions, and not the original active context. Is that correct?

**Comment by gkellogg** [comment]
(I believe 5.1.1 is the step you meant).

I think 5.1.1 should check _result_ instead of _active context_, so the paragraph could read as follows:

| If _override protected_ is `false` and _result_ contains any *protected term definitions*, an `invalid context nullification` has been detected and processing is aborted.

cc/ @dlongley @pchampin

**Comment by gkellogg** [comment]
@iherman Should this be marked as Editorial, given our general policy about tests winning out over algorithm text?

**Comment by dlongley** [comment]
I imagine the next step (5.1.2) may need to talk about setting `result` back to an initial active context instead of saying it's being initialized as well. I do think it's clearly a typo/editorial.

**Comment by iherman** [comment]
> @iherman Should this be marked as Editorial, given our general policy about tests winning out over algorithm text?

I do not think so. If my understanding is correct, the normative description of the algorithm changes, which may, potentially, affect other implementations (even though the change is not far-reaching).

**Comment by gkellogg** [comment]
Summary:

Update the [Context Processing Algorithm](https://www.w3.org/TR/json-ld11-api/#algorithm) steps 5.1.1 and 5.1.2 to look for protected term definitions in _result_ not _active context_ and to initialize _result_ to an initial active context as follows:

> 5.1.1) If _override protected_ is `false` and _result_ contains any [protected](https://www.w3.org/TR/json-ld11-api/#dfn-protected) [term definitions](https://www.w3.org/TR/json-ld11/#dfn-term-definition), an [invalid context nullification](https://www.w3.org/TR/json-ld11-api/#dom-jsonlderrorcode-invalid-context-nullification) has been detected and processing is aborted.
> 5.1.2) Set _result_ as a new initial [active context](https://www.w3.org/TR/json-ld11/#dfn-active-context) as defined above, and setting both [base IRI](https://www.w3.org/TR/json-ld11-api/#dfn-context-base-iri) and [original base URL](https://www.w3.org/TR/json-ld11-api/#dfn-original-base-url) to the value of [original base URL](https://www.w3.org/TR/json-ld11-api/#dfn-original-base-url) in _active context_, and, if _propagate_ is `false`, [previous context](https://www.w3.org/TR/json-ld11-api/#dfn-previous-context) in _result_ to the previous value of _result_.
